### PR TITLE
Added new tests to test llvm 14 and llvm 15 system configurations

### PR DESCRIPTION
**Pull request by bhavanijayakumaran**
As part of the llvm 15 upgrade added 2 new tests.
Issue: https://github.com/Cray/chapel-private/issues/4814

### CODE REVIEW

File: util/cron/test-linux64-llvm15.bash [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+#
+# Test default configuration on examples only, on linux64, with llvm 15
+
+export OFFICIAL_SYSTEM_LLVM=true
+source /data/cf/chapel/setup_system_llvm.bash 15
```
Comment (mppf) [inline util/cron/test-linux64-llvm15.bash:6]: @riftEmber do we have this script working with the new Spack build of LLVM 15? Or does this need to do something else?

**Review by mppf — APPROVED** [review]
Looks good after fixing the typo

File: util/cron/test-linux64-llvm14.bash [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+source $CWD/common.bash
+
+clang_version=$(clang -dumpversion)
+if [ "$clang_version" != "14.0.0" ]; then
+  echo "Wrong clang version"
+  echo "Expected Version: 14.0.0 Actual Version: $clang_version"
+  exit 2
+fi
+
+export CHPL_NIGHTLY_TEST_CONFIG_NAME="linux64-llvm15"
```
[the remark this answers is not in the document]
Comment (bhavanijayakumaran) [reply util/cron/test-linux64-llvm14.bash:18]: Done.

File: util/cron/test-linux64-llvm15.bash (hunk 1 above)
Comment (riftEmber) [reply util/cron/test-linux64-llvm15.bash:6]: Not using spack. My understanding (which may have been incorrect) was that Bhavani and I were planning to work together on hooking up the Spack build of LLVM 15 to be used before these new configs went in. I didn't actually realize there even was a manual build of LLVM 15 here. In any case I can just change it to use the spack build after the fact.

**Pull request merged**